examples/pxScene2d/src/pxShaderResource.cpp

```diff
@@ -0,0 +1,1433 @@
+{
+#if 0
+  pxArchive* archive = (pxArchive*)archiveRef.getPtr();
+  rtString status = "resolve";
+
+  rtError loadShaderSuccess = RT_FAIL;
+
+  // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+  if(mFragmentSrc.length() == 0)
+  {
+    if ((NULL != archive) && (RT_OK == archive->getFileData(mFragmentUrl, mFragmentSrc)))
+    {
+      loadShaderSuccess = RT_OK;
+    }
+    else
+    {
+      loadShaderSuccess = RT_RESOURCE_NOT_FOUND;
+      rtLogError("Could not load shader file from archive %s.", mUrl.cString());
+    }
+  }
+  else
+  {
+    // We have SHADER source as a string ...
+    loadShaderSuccess = RT_OK;
+  }
+  // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+  // CREATE SHADER PROGRAMS
+
+  if (loadShaderSuccess == RT_OK)
+  {
+    double startDecodeTime = pxMilliseconds();
+    loadShaderSuccess = RT_OK;
+
+    const char* vtxCode = mVertexSrc.length() > 0 ? (const char*) mVertexSrc.data() : vShaderText;
+
+    if(shaderProgram::initShader( vtxCode, (const char*) mFragmentSrc.data() ) != RT_OK)
+    {
+      rtLogError("FATAL: Shader error: %s \n", mCompilation.cString() );
+      
+      setLoadStatus("glError", mCompilation.cString() );
+      
+      gUIThreadQueue->addTask(onDownloadCanceledUI, this, (void*)"reject");
+    }
+
+    double stopDecodeTime = pxMilliseconds();
+    setLoadStatus("decodeTimeMs", static_cast<int>(stopDecodeTime-startDecodeTime));
+  }
+  else
+  {
+    loadShaderSuccess = RT_RESOURCE_NOT_FOUND;
+    rtLogError("Could not load shader file %s.", mFragmentUrl.cString());
+  }
+
+  // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+  if ( loadShaderSuccess != RT_OK)
+  {
+    rtLogWarn("shader load failed"); // TODO: why?
+    if (loadShaderSuccess == RT_RESOURCE_NOT_FOUND)
+    {
+      setLoadStatus("statusCode",PX_RESOURCE_STATUS_FILE_NOT_FOUND);
+    }
+    else
+    {
+      setLoadStatus("statusCode", PX_RESOURCE_STATUS_DECODE_FAILURE);
+    }
+
+    // Since this object can be released before we get a async completion
+    // We need to maintain this object's lifetime
+    // TODO review overall flow and organization
+    AddRef();
+
+    if (gUIThreadQueue)
+    {
+      gUIThreadQueue->addTask(onDownloadCanceledUI, this, (void*)"reject");
+    }
+    //mTexture->notifyListeners( mTexture, RT_FAIL, errorCode);
+  }
+  else
+  {
+    mFragmentSrc.term(); // Dump the source data...
+    mVertexSrc.term();   // Dump the source data...
+
+    setLoadStatus("statusCode",0);
+    // Since this object can be released before we get a async completion
+    // We need to maintain this object's lifetime
+    // TODO review overall flow and organization
+    AddRef();
+    if (gUIThreadQueue)
+    {
+      gUIThreadQueue->addTask(onDownloadCompleteUI, this, (void *) "resolve");
+    }
+  }
+#endif // 0
```

Comment:
Let's log a warning if we are trying to load a shader resource from an archive since it's not supported at this time.  This will help debug the issue quickly in the future.

Reply:
Done

Reply:
Done